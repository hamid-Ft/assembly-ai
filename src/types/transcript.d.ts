
export interface Entities {
  entity_type: string;
  text: string;
  start?: number;
  end?: number;
}
export interface Words {
  text: string;
  start: number;
  end: number;
  confidence: number;
  speaker: string;
}
export interface Utterance {
  confidence: number;
  end: number;
  speaker: string;
  start: number;
  text: string;
  words: Words[];
}
export interface SentimentAnalysisResults {
  confidence: number;
  text: string;
  sentiment: 'POSITIVE' | 'NEGATIVE' | 'NEUTRAL';
}

export interface Label {
  relevance: number;
  label: string;
}

export interface Result {
  text: string;
  labels: Label[];
  timestamp: {
    start: number;
    end: number;
  };
}

export interface Summary {
  topic: number;
}

export interface IabCategoriesResult {
  status: 'success' | 'unavailable';
  results: Result[];
  summary: Summary;
} 
export interface StatusProps {
  isLoading: boolean;
  status: string;
}


export interface Transcript {
  id: string;
  status?: 'processing' | 'queued' | 'completed' | 'error';
  text?: string;
  sentiment_analysis_results?: SentimentAnalysisResults[];
  entities?: Entities[];
  utterances?: Utterance[];
  iab_categories_result?: IabCategoriesResult;
}
