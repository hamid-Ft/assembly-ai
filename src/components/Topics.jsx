/* eslint-disable react/prop-types */
const Topics = ({ transcript }) => {
  return (
    <div>
      <h3>Topics Detected: </h3>
      <div>
        {Object.keys(transcript.iab_categories_result.summary)
          .filter(
            (topic) => transcript.iab_categories_result.summary[topic] > 0.6
          )
          .map((topic) => (
            <span
              key={transcript.iab_categories_result.summary[topic]}
              style={{ fontSize: "small", backgroundColor: "blueviolet" }}>
              {topic.split(">").pop()}
            </span>
          ))}
      </div>
    </div>
  );
};

export default Topics;
