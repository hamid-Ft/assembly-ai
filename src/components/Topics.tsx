import React from "react";

const Topics = ({ transcript }) => {
  return (
    <div>
      <h3>Topics Detected: </h3>
      <div>
        {Object.keys(transcript.iab_categories_result.summary)
          .filter(
            (topic) => transcript.iab_categories_result.summary[topic] > 0.7
          )
          .map((topic) => (
            <span
              key={transcript.iab_categories_result.summary[topic]}
              style={{
                fontSize: "small",
                backgroundColor: "blueviolet",
                paddingInline: "0.25rem",
                borderRadius: "10px",
                marginInline: "0.5  rem",
              }}>
              {topic.split(">").pop()}
            </span>
          ))}
      </div>
    </div>
  );
};

export default Topics;
