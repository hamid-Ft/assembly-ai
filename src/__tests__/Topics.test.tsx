import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Topics from '../components/Topics';
import { Transcript } from '../types/transcript';

describe('Topics component', () => {
    test('renders Topics component', () => {
        // Arrange
        const mockTranscript: Transcript = {
            id: '1',
            iab_categories_result: {
                status: 'success',
                results: [],
                summary: { topic: 1 },
            },
        };

        // Act
        render(<Topics transcript={mockTranscript} />);

        // Assert
        expect(screen.getByText(/Topics Detected:/i)).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        // Arrange
        const mockTranscript: Transcript = {
            id: '1',
            iab_categories_result: {
                status: 'success',
                results: [],
                summary: { topic: 1 },
            },
        };

        // Act
        const { asFragment } = render(<Topics transcript={mockTranscript} />);

        // Assert
        expect(asFragment()).toMatchSnapshot();
    });
});
