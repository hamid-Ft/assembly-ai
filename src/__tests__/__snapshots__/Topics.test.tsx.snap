// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topics component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Topics Detected: 
    </h3>
    <div>
      <span
        style="font-size: small; background-color: blueviolet; padding-inline: 0.25rem; border-radius: 10px; margin-inline: 0.5  rem;"
      >
        topic
      </span>
    </div>
  </div>
</DocumentFragment>
`;
